cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -1912,6 +1912,16 @@ module BarrierGuard<guardChecksSig/3 guardChecks> {
       g.controls(result.getBasicBlock(), edge)
     )
   }
+
+  /** Gets an indirect expression node that is safely guarded by the given guard check. */
+  IndirectExprNode getAnIndirectBarrierNode() {
```

Comment:
Do we want to expose the indirection index here as an argument of `getAnIndirectBarrierNode`?

Reply:
We could do that, yeah. And then implement a `getAnIndirectBarrierNode` with no parameters as `getAnIndirectBarrierNode(_)`?

Reply:
Yup.

Reply:
Fixed in 2d98fb7cf1b20ece014ae132416c5b6f2addb03a.